src/main/java/taxi/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,44 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/register");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse,
+                         FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        HttpSession httpSession = req.getSession();
+        Long id = (Long) httpSession.getAttribute("driver_id");
+        if (id == null) {
+            if (allowedUrls.contains(req.getServletPath())) {
```

Comment:
is there a way to make it bettter?
HINT:
if (driverId != null || allowedUrls.contains(req.getServletPath()))
So if we have authorized user (not null), or we want to go to allowed pages for everyone - we can execute doFilter.